atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -152,14 +152,14 @@ public void drop() {
             // If table does not exist, do nothing
         }
 
-        dropTableMetadataAndClearTableSizeCache();
+        clearTableSizeCacheAndDropTableMetadata();
     }
 
-    private void dropTableMetadataAndClearTableSizeCache() {
+    private void clearTableSizeCacheAndDropTableMetadata() {
+        TableSizeCache.clearCacheForTable(tableRef);
```

Comment:
getting a cache hit after dropping the table will be of no use as any query will fail:

eg:

`[NODE 1 THREAD 1] Create table A
[NODE 1 THREAD 1] Delete table A
[NODE 1 THREAD 1] Clear cache
[NODE 1 THREAD 2] Query table A  --> fails (RuntimeException)
[NODE 1 THREAD 1] Delete Table A from _metadata
[NODE 2 THREAD 1] Create table B
[NODE 1 THREAD 2] Query table A`

Locking the cache is safe, but I dont think we are at risk here.